lib/resourcebuilder/batch.go

```diff
@@ -46,15 +46,26 @@ func checkJobHealth(ctx context.Context, client batchclientv1.JobsGetter, job *b
 	}
 
 	// Since we have filled in "activeDeadlineSeconds",
-	// the Job will 'Active == 0' if and only if it exceeds the deadline.
+	// the Job will 'Active == 0' if and only if it exceeds the deadline or if the update image could not be pulled.
 	// Failed jobs will be recreated in the next run.
-	if j.Status.Active == 0 && j.Status.Failed > 0 {
-		reason := "DeadlineExceeded"
-		message := "Job was active longer than specified deadline"
-		if len(j.Status.Conditions) > 0 {
-			reason, message = j.Status.Conditions[0].Reason, j.Status.Conditions[0].Message
+	if j.Status.Active == 0 {
+		klog.V(4).Infof("No active pods for job %s in namespace %s", job.Name, job.Namespace)
+		// If there is more than one failed job pod then get the cause for failure
+		if j.Status.Failed > 0 {
+			reason := "DeadlineExceeded"
+			message := "Job was active longer than specified deadline"
+			if len(j.Status.Conditions) > 0 {
+				reason, message = j.Status.Conditions[0].Reason, j.Status.Conditions[0].Message
+			}
+			return false, fmt.Errorf("deadline exceeded, reason: %q, message: %q", reason, message)
+		}
+
+		// When the update image cannot be pulled then the pod is not marked as Failed, but the status condition is set
+		// after the job deadline is exceeded.
+		if len(j.Status.Conditions) > 0 && j.Status.Conditions[0].Reason == "DeadlineExceeded" {
```

Comment:
I see we were using `Conditions[0]` before as well, but ideally we'd be looking for specific condition types.  Because maybe sometime in the future, the job controller adds a `OptimisticThatTheNextAttemptWillPass=True` condition, and we don't want to use that when we complain about it being broken so far.